### DISCUSSION TITLE
Phantom registration is not avilable

### DISCUSSION THREAD

**Discussion by HiunIce** [PlusToolkit/PlusLib#958] [Q&A]
Hi, teams

I encountered the following question in the spatial calibration step when I using the software to calibrate the ultrasound probe. 
Although there is an orange prompt, I still don't know how to fix it.
Could anyone give me some advice? 

Thanks a lot!
Best regards.

The error in software here:
![image](https://user-images.githubusercontent.com/36978849/173271969-7c21069b-2b50-42e8-bc92-365171248436.png)

The error Log:

![image](https://user-images.githubusercontent.com/36978849/173272335-c0efa319-f0b1-4d96-bc17-c0da258fabed.png)

My xml data file:

![image](https://user-images.githubusercontent.com/36978849/173272401-08ab6f29-b11f-451a-a2f2-9c603ddcd1ce.png)

**Comment by Sunderlandkyl** [comment]
Can you attach the full config file and log as txt files?

**Comment by HiunIce** [comment]
The attachment is my log file and setting xml file. @Sunderlandkyl 

Thanks a lot!

[fcal error.zip](https://github.com/PlusToolkit/PlusLib/files/8892530/fcal.error.zip)
[061322_104319_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/8892533/061322_104319_PlusLog.txt)

**Reply by Sunderlandkyl** [reply]
The probe coordinate system from the device is given as "USProbe", while the coordinate system in the rest of the config file is "Probe",
Try replacing all instances of ```ProbeCoordinateFrame="Probe"``` with ```ProbeCoordinateFrame="USProbe"```.

**Reply by Sunderlandkyl** [reply]
Also ProbeToTracker with USProbeToTracker.

**Reply by HiunIce** [reply]
Thanks!
I'll try it once I get back to the lab tomorrow, I think that would work

**Reply by HiunIce** [reply]
> Also ProbeToTracker with USProbeToTracker.

Sorry, but it fails with the same errors.

I tried in two ways.

A:  change "probe" to us probe as you written above.
B:  keep Probe Tags, and change all USProbe to Probe.

Both method A & B failed.
The attachment is xml and log file for method A.

Could you give me more suggestions? 
Or show me an example of NDI & Video Capture Card (for us probe) Setting?
Thanks!
Best regards.


[error type2.zip](https://github.com/PlusToolkit/PlusLib/files/8901125/error.type2.zip)

**Reply by Sunderlandkyl** [reply]
Looks like you also need to specify ToolReferenceFrame="Tracker" in "TrackerDevice".

**Reply by HiunIce** [reply]
I am terribly sorry for I failed again.  
The 【ToolReferenceFrame="Tracker" is】 added as below 

  <DataCollection StartupDelaySec="1">
    <DeviceSet Name="temporalR" Description="Mmf framegrabber image acquisition + NDI Polaris for tracking" />
    <Device

And I searched the source code and found the bug occurs in the class vtkPlusVisualizationController.

So I tried to compile this software.

Hence I build [vtkAddon->IGSIO->PlusLib] with VTK9.1 & itkITK-4.13
But failed when building release version of IGSIO (debug version is ok)
The details are following.

Or could you please give me builded IGSIO?

And here is the app logs and xml, hope for your replying

[061522_175231_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/8918608/061522_175231_PlusLog.txt)
[temporalRE2.zip](https://github.com/PlusToolkit/PlusLib/files/8918617/temporalRE2.zip)

Thank you!
Best regards.


Lib Build list:

vtkAddon-->success

IGSIO --> debug ok, release failed
(bugs:
 MDd_dynamic...
itkvnl-4.13.lib  valid __imp__invalid_parameter
)


PlusLib:  Errors
CMake Error at CMakeLists.txt:116 (STRING):
  STRING sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.


CMake Error at CMakeLists.txt:117 (STRING):
  STRING sub-command REPLACE requires at least four arguments.


CMake Error at src/PlusCommon/CMakeLists.txt:38 (GET_PROPERTY):
  GET_PROPERTY could not find TARGET igtlioConverter.  Perhaps it has not yet
  been created.

**Reply by Sunderlandkyl** [reply]
What platform were you building on?

Could you provide a link to the issue you found in vtkPlusVisualizationController?

**Reply by HiunIce** [reply]
I am building it on windows 10.

what i found in the class is to get the phantom matrix from files.

It does not on my phone.
I am going to build the app on another computer，hope it works.

(^ ^)

**Reply by jonortega77** [reply]
any update with this problem? I am facing the same problem and I am not able to find the solution.

Any help would be appreciated !

[config_file.txt](https://github.com/user-attachments/files/17585427/config_file.txt)
[logs_error_spatial_cal.txt](https://github.com/user-attachments/files/17585429/logs_error_spatial_cal.txt)
